### PR TITLE
Add release notes covering configuration cache improvements in 7.5

### PR DESCRIPTION
**Pull request by mlopatkin**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -181,6 +181,37 @@ This allows configuration of PMD to run its analysis on more than one thread.
 
 See the [documentation](userguide/pmd_plugin.html#sec:pmd_conf_threads) for more information.
 
+### Configuration cache improvements
+
+#### Modification of system properties at the configuration time
```
Comment (eskatos) [inline subprojects/docs/src/docs/release/notes.md:186]: This is about a bug fixed. I wonder if we should mention it in the release notes, probably not.

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -181,6 +181,37 @@ This allows configuration of PMD to run its analysis on more than one thread.
...
+#### Running external processes at the configuration time
+
+Running external processes at the configuration time with Java or Groovy APIs or with `exec` and `javaexec` Gradle APIs is
+now [disallowed](userguide/configuration_cache.html#config_cache:requirements:external_processes) if the configuration cache is enabled.
+A [new Provider-based API](javadoc/org/gradle/api/provider/ProviderFactory.html#exec-org.gradle.api.Action-) is
+now available to obtain an output of the external process in the configuration-cache-compatible way.
+If a more complex interaction with the external process is necessary, then a custom [`ValueSource`](javadoc/org/gradle/api/provider/ValueSource.html) implementation
+can be used. The injected [`ExecOperations`](javadoc/org/gradle/process/ExecOperations.html) service should be used to run the external process.
+
+#### Accessing the environment in the ValueSource implementation no longer introduces additional build configuration inputs
```
Comment (eskatos) [inline subprojects/docs/src/docs/release/notes.md:201]: I would move this item down, file system reads and external process execution are the most common use case. Custom value sources is more advanced.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -181,6 +181,37 @@ This allows configuration of PMD to run its analysis on more than one thread.
...
+#### Modification of system properties at the configuration time
+
+Changes to system properties made at the configuration time are now also applied when the configuration cache entry is reused.
+This includes adding a property, modifying or removing an existing one.
+The types used for property keys and values have to be [supported by the configuration cache](userguide/configuration_cache.html#config_cache:requirements:disallowed_types).
+
+#### Running external processes at the configuration time
+
+Running external processes at the configuration time with Java or Groovy APIs or with `exec` and `javaexec` Gradle APIs is
+now [disallowed](userguide/configuration_cache.html#config_cache:requirements:external_processes) if the configuration cache is enabled.
```
Comment (eskatos) [inline subprojects/docs/src/docs/release/notes.md:195]: We should say that before we ignored them and that was a correctness issue.

File: subprojects/docs/src/docs/release/notes.md [hunk 4]
```diff
@@ -181,6 +181,37 @@ This allows configuration of PMD to run its analysis on more than one thread.
...
+Initially, the [`ValueSource`](javadoc/org/gradle/api/provider/ValueSource.html) implementations were subject to the same input tracking rules as the rest of the build logic.
+Reading a file, an environment variable, or a system property would make this a build configuration input.
+Any change to the input would invalidate the configuration cache, even if the value of the `ValueSource` doesn't change.
+
+Now, reading a file, an environment variable, or a system property is no longer tracked inside the implementation of the `ValueSource`.
+The value of the `ValueSource` is still recomputed each time the build runs, and the configuration cache entry is only invalidated if the value changes.
+
+#### Files read at the configuration time become build configuration inputs
+
+Files read at the configuration time with `FileInputStream` or some Kotlin APIs now become build configuration inputs.
```
Comment (eskatos) [inline subprojects/docs/src/docs/release/notes.md:212]: We should say that before we ignored them and that was a correctness issue.

File: subprojects/docs/src/docs/release/notes.md (hunk 1 above)
Comment (mlopatkin) [reply subprojects/docs/src/docs/release/notes.md]: It's borderline, IMO, but I don't have a strong enough opinion about this one.

File: subprojects/docs/src/docs/release/notes.md (hunk 3 above)
Comment (mlopatkin) [reply subprojects/docs/src/docs/release/notes.md:189]: Done

File: subprojects/docs/src/docs/release/notes.md (hunk 2 above)
Comment (mlopatkin) [reply subprojects/docs/src/docs/release/notes.md:202]: Done

File: subprojects/docs/src/docs/release/notes.md (hunk 4 above)
Comment (mlopatkin) [reply subprojects/docs/src/docs/release/notes.md]: Done

**Review by mlopatkin — COMMENTED** [review]
Thanks, I've applied your suggestions.
I'll check with the support team if it is okay to merge, or should I apply the changes to the GDoc.

**Comment by mlopatkin** [comment]
GDoc is the preferred way at this point.

**Pull request closed without merging**